lang/syn/src/parser/program/instructions.rs

```diff
@@ -91,3 +93,34 @@ pub fn parse_args(method: &syn::ItemFn) -> ParseResult<(IxArg, Vec<IxArg>)> {
 
     Ok((ctx, args))
 }
+
+pub fn parse_return(method: &syn::ItemFn) -> ParseResult<IxReturn> {
+    match method.sig.output {
+        syn::ReturnType::Type(_, ref ty) => {
```

Comment:
an error here would be nice that hardcodes (with `.to_string`) that the return type must be of the form `Result<()>` or `Result<T>`

the return code (and possibly even other code) depends on this exact return type and you can't use aliases like `type MyResult = Result<()>`

Reply:
There is still some tests returning `ProgramResult`, this handles return types that don't confirm to `Result<()>` or `Result<T>` by defaulting to assuming it returns the unit type, these might break I think? I can commit it anyway and let CI figure it out

Reply:
yea I cant think of anything that can actually go wrong with your code defaulting to no return type. so no need for that extra check i think